utils/pluggable.hh

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#pragma once
+
+#include <stdexcept>
+#include <optional>
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/shared_ptr.hh>
+
+#include "seastarx.hh"
+#include "utils/phased_barrier.hh"
+#include "utils/on_internal_error.hh"
+
+namespace utils {
+
+/// This class provides an interface for pluggable async services
```

Comment:
This documentation is circular

Reply:
Improved doc comment in v3 ([8ae8275](https://github.com/scylladb/scylladb/commit/8ae8275f17e252ae3d43aa6440213c3e21996050))

Reply:
Fixed in v2